CONCEPTS.md

```diff
@@ -1,19 +1,41 @@
-   * precision - define the precision of the calculations in your model
-   * dataflow/resource reuse - control parallel or serial model implementations with varying levels of pipelining
 
-The hls4ml package enables fast prototyping of a machine learning algorithm implementation in FPGAs,
-greatly reducing the time to results and giving the user intuition for how to best design a machine learning algorithm for their application while balancing performance, resource utilization and latency requirements.
\ No newline at end of file
+With this in mind, let's take a look at how hls4ml helps to achieve such a goal. First, it's important to realize the architecture differences between an FPGA and a CPU or GPU. An FPGA can be specifically programmed to do a certain task, in this case evaluate neural networks given a set of inputs, and as such can be highly optimized for the task, with tricks like pipelining and parallel evaluation. However, this means dynamic remapping while running isn't really a possibility. FPGAs also often come at a comparatively low power cost with respect to CPUs and GPUs. This allows hls4ml to build HLS code from compressed neural networks that results in predictions on the microsecond scale for latency. The hls4ml tool saves the time investment needed to convert a neural network to a hardware design language or even HLS code, thus allowing for rapid prototyping.
+
+
+## How it Works
+
+<br>
+<div style="text-align: center;"><img src="img/nn_map_paper_fig_2.png" width="400" /></div>
+<br><br>
+
+Consider a multi-layered neural network. At each neuron in a layer $m$ (containing $N_m$ neurons), we calculate an output value (part of the output vector $\mathbf{x}_m$ of said layer) using the sum of output values of the previous layer multiplied by independent weights for each of these values and a bias value. An activation function is performed on the result to get the final output value for the neuron. Representing the weights as a $N_m \times N_{m-1}$ matrix $\mathbf{W}_{m,m-1}$, the bias values as $\mathbf{b}_m$, and the activation function as $g_m$, we can express this compactly as $\mathbf{x}_m = g_m(\mathbf{W}_{m,m-1}\mathbf{x}_{m-1}+\mathbf{b}_m)$. With hls4ml, each layer of output values is calculated independently in sequence, using pipelining to speed up the process by accepting new inputs after an initiation interval. The activations, if nontrivial, are precomputed. 
+
+To ensure optimal performance, the user can control aspects of their model, principally:
+
+   * **Size/Compression** - Though not explicitly part of the hls4ml package, this is an important optimization to efficiently use the FPGA resources
+   * **Precision** - Define the precision of the calculations in your model
+   * **Dataflow/Resource Reuse** - Control parallel or serial model implementations with varying levels of pipelining
+
```

Comment:
I think here we could add a link from the Precision bullet point to the PROFILING.md page. 
And maybe add another bullet afterwards like:

- Quantization Aware Training - achieve best performance at low precision with tools like QKeras, and benefit automatically during inference with hls4ml parsing of QKeras models

(It's kind of in the same vein as the Size/Compression bullet: something outside of hls4ml that benefits the inference performance)

Reply:
Thanks for the suggestion, I'll add that